RubeesEat/Model/DB/DbBillRepository.cs

```diff
@@ -0,0 +1,254 @@
+using System.Collections.Immutable;
+using System.Data;
+using System.Data.Common;
+using Microsoft.Data.SqlClient;
+
+namespace RubeesEat.Model.DB;
+
+public class DbBillRepository(IDbConnectionFactory connectionFactory) : IBillRepository
+{
+    public IReadOnlyList<Bill> GetAll()
+    {
+        using var connection = connectionFactory.CreateDbConnection();
+        using var command = connection.CreateCommand();
+
+        command.CommandText = $"""
+                                 SELECT
+                                       B.BillID,
+                                       B.Date,
+                                       B.Description,
+                                       E.EntryLineID,
+                                       E.Amount,
+                                       P.PersonID,
+                                       P.FirstName,
+                                       P.LastName
+                                   FROM
+                                       Bills B
+                                   INNER JOIN
+                                       EntryLines E ON B.BillID = E.BillID
+                                    INNER JOIN
+                                       Persons P ON P.PersonID = E.PersonID
+                                   ORDER BY B.Date DESC, B.BillID;
+                               """;
+        command.CommandType = CommandType.Text;
+        connection.Open();
+        using var reader = command.ExecuteReader();
+        return CreateBillsFromReader(reader);
+    }
+
+    public void Add(Bill bill)
+    {
+        using (var connection = connectionFactory.CreateDbConnection())
+        {
+            connection.Open();
+            using (var command = connection.CreateCommand())
+            {
+                SqlTransaction transaction;
+                
+                transaction = (SqlTransaction)connection.BeginTransaction();
```

Comment:
What kind of transaction are starting here and is it the one we want?

Reply:
Do you mean one with Isolation level unspecified?

Reply:
I am guessing we only need Isolationlevel Read Committed, so that other queries cannot read our unfinished transaction. It seems like repeatable reads and phantom reads shouldn't be a problem for now, since we don't read lines multiple times, or do the same query in the same function call